### DISCUSSION TITLE
Geolocation Custom UIType

### DISCUSSION THREAD

**Discussion by apcloic** [YetiForceCompany/YetiForceCRM#16602] [How to]
Hi,

I've created a custom UIType for geolocation in order to locate user's position.
Of course, user needs to approve like any other site requesting for coordinates.
For those interested, you will find the required files and modifications below.

Unfortunately, changes made in /app and /public_html are not upgrade safe.
Also, it requires to enable  $LOAD_CUSTOM_FILES and disable  $MINIFY_JS in CONFIG.
If someone know how to improve the code and don't touch core file, let us know.

Regards,

![geolocation_uitype](https://user-images.githubusercontent.com/2930560/196432058-60ef85ca-06bf-47ab-9f39-3cac1c8c39b3.gif)

[geolocation_custom_uitype.zip](https://github.com/YetiForceCompany/YetiForceCRM/files/9811097/geolocation_custom_uitype.zip)

**Comment by mariuszkrzaczkowski** [comment]
Interesting idea, but what does this field give you?
I can see that it only directs to the XML file but functionally what is it supposed to do?

**Reply by apcloic** [reply]
@mariuszkrzaczkowski 
Basically, I only need that field to save user's GPS coordinates (for security officers).
As I was learning how to create a new UIType, I also tried to see if I could make it "clickable".
So the XML link is more a proof-of-concept than something really useful. :)
I think I could open a map instead of an XML :

````
http://www.openstreetmap.org/?mlat=xx.xxxxxx&mlon=x.xxxxxxx&zoom=15

````

**Reply by mariuszkrzaczkowski** [reply]
Ideally, it would integrate with the built-in map

**Reply by apcloic** [reply]
@mariuszkrzaczkowski thanks a lot for your feedback.
You're right, it's better to integrate with built-in map, I will try later  :)

**Comment by mariuszkrzaczkowski** [comment]
There is a new uitype on the development version similar to what you proposed,
It's not finished yet, lacks map integration and some other improvements.
![image](https://user-images.githubusercontent.com/10198654/203750672-24dfc133-8b0c-4199-ba27-e962ca5dfd74.png)
![image](https://user-images.githubusercontent.com/10198654/203750682-b1f88808-2afa-46a6-ae51-03a4807c0e34.png)
![image](https://user-images.githubusercontent.com/10198654/203750697-d581ab73-4328-4666-aec0-b59ad0da9425.png)
![image](https://user-images.githubusercontent.com/10198654/203750710-614f5b8b-dbfa-4a23-9708-e3c4115c52e0.png)
![image](https://user-images.githubusercontent.com/10198654/203750725-58ab498b-d538-49db-b216-386efd208c42.png)

**Reply by mariuszkrzaczkowski** [reply]
https://gitdeveloper.yetiforce.com/index.php?module=Locations&view=Edit&parent=76&mid=163&record=1459
